extensions/amp-megaphone/0.1/amp-megaphone.js

```diff
@@ -0,0 +1,234 @@
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Embeds a Megaphone podcast
+ *
+ * Example:
+ * <code>
+ * <amp-megaphone
+ *   height=166
+ *   data-episode="DEM6617440160"
+ *   data-light="true"
+ *   layout="fixed-height">
+ * </amp-megaphone>
+ */
+
+import {Layout} from '../../../src/layout';
+import {addParamsToUrl} from '../../../src/url';
+import {dict} from '../../../src/utils/object';
+import {getData, listen} from '../../../src/event-helper';
+import {isObject} from '../../../src/types';
+import {removeElement} from '../../../src/dom';
+import {startsWith} from '../../../src/string';
+import {tryParseJson} from '../../../src/json';
+import {userAssert} from '../../../src/log';
+
+class AmpMegaphone extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {?Function} */
+    this.unlistenMessage_ = null;
+
+    /** @private {boolean} */
+    this.isPlaylist_ = false;
+
+    /** @private {string} */
+    this.baseUrl_ = '';
+  }
+
+  /**
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+    this.preconnect.url(this.baseUrl_, opt_onLayout);
+    this.preconnect.url('https://assets.megaphone.fm', opt_onLayout);
+    this.preconnect.url('https://megaphone.imgix.net', opt_onLayout);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return (
+      layout == Layout.FIXED ||
+      layout == Layout.FIXED_HEIGHT ||
+      layout == Layout.FILL ||
+      layout == Layout.NODISPLAY
+    );
+  }
+
+  /** @override */
+  buildCallback() {
+    this.updateBaseUrl_();
+  }
+
+  /** @override */
+  mutatedAttributesCallback(mutations) {
+    const playlist = mutations['data-playlist'];
+    const episode = mutations['data-playlist'];
+    if (playlist !== undefined || episode !== undefined) {
+      this.updateBaseUrl_();
+    }
+
+    const light = mutations['data-light'];
+    const sharing = mutations['data-sharing'];
+    const episodes = mutations['data-episodes'];
+    const start = mutations['data-start'];
+    const tile = mutations['data-tile'];
+    if (
+      light !== undefined ||
+      sharing !== undefined ||
+      episodes !== undefined ||
+      start !== undefined ||
+      tile !== undefined
+    ) {
+      if (this.iframe_) {
+        this.iframe_.src = this.getIframeSrc_();
+      }
+    }
+  }
+
+  /**@override*/
+  layoutCallback() {
+    const height = this.element.getAttribute('height');
+    const iframe = this.element.ownerDocument.createElement('iframe');
+
+    iframe.setAttribute('frameborder', 'no');
+    iframe.setAttribute('scrolling', 'no');
+
+    iframe.src = this.getIframeSrc_();
+
+    this.unlistenMessage_ = listen(
+      this.win,
+      'message',
+      this.handleMegaphoneMessages_.bind(this)
+    );
+
+    this.applyFillContent(iframe);
+    iframe.height = height;
```

Comment:
Why do we need to give the iframe a height? Can this be `100%`? This is not clear, so at least add a comment.